lib/converter/policy.go

```diff
@@ -25,19 +25,22 @@ import (
 type PolicyConverter struct{}
 
 // ConvertMetadataToKey converts a PolicyMetadata to a PolicyKey
-func (p PolicyConverter) ConvertMetadataToKey(m unversioned.ResourceMetadata) model.Key {
+func (p PolicyConverter) ConvertMetadataToKey(m unversioned.ResourceMetadata) (model.Key, error) {
 	pm := m.(api.PolicyMetadata)
 	k := model.PolicyKey{
 		Name: pm.Name,
 	}
-	return k
+	return k, nil
 }
 
 // ConvertAPIToKVPair converts an API Policy structure to a KVPair containing a
 // backend Policy and PolicyKey.
-func (p PolicyConverter) ConvertAPIToKVPair(a unversioned.Resource) *model.KVPair {
+func (p PolicyConverter) ConvertAPIToKVPair(a unversioned.Resource) (*model.KVPair, error) {
 	ap := a.(api.Policy)
-	k := p.ConvertMetadataToKey(ap.Metadata)
+	k, err := p.ConvertMetadataToKey(ap.Metadata)
```

Comment:
Converting from Metadata to Key should always succeed tbh.  It would be nice to just get rid of that error from all of the interfaces in both client and k8s.